src/Opdex.Platform.Infrastructure/Data/Handlers/Tokens/SelectTokensWithFilterQueryHandler.cs

```diff
@@ -35,14 +36,13 @@ public class SelectTokensWithFilterQueryHandler : IRequestHandler<SelectTokensWi
                 t.{nameof(TokenEntity.TotalSupply)},
                 t.{nameof(TokenEntity.CreatedBlock)},
                 t.{nameof(TokenEntity.ModifiedBlock)},
-                ts.{nameof(TokenSummaryEntity.PriceUsd)},
-                ts.{nameof(TokenSummaryEntity.DailyPriceChangePercent)}
+                AVG(ts.{nameof(TokenSummaryEntity.PriceUsd)}) AS {nameof(TokenSummaryEntity.PriceUsd)},
+                AVG(ts.{nameof(TokenSummaryEntity.DailyPriceChangePercent)}) AS {nameof(TokenSummaryEntity.DailyPriceChangePercent)}
```

Comment:
This is going to average all of the results prices, from all tokens.

it will require either a new `SELECT` internally here or another query altogether to specifically average a one token across markets. 

I think we should just return the price based on the specific market, for market tokens.

We should probably split these queries out for Market/Global tokens and not allow filtering based on averaged token summary data for global tokens.

Reply:
The `GROUP BY t.{nameof(TokenEntity.Id)}` should cause the average to be per token